### PR TITLE
LIME-419 - Third Party response times recorded in metrics

### PR DESCRIPTION
**Pull request by MMcC-GDS**
### What changed

- Fraud Check duration metric added to AWS Cloudwatch.
- PEP Check duration metric added to AWS Cloudwatch.

### Issue tracking

- [LIME-419](https://govukverify.atlassian.net/browse/LIME-419)

[LIME-419]: https://govukverify.atlassian.net/browse/LIME-419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

image

### CODE REVIEW

**Review by ElliotMurphyGDS — DISMISSED** [review]

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java [hunk 1]
```diff
@@ -128,6 +129,9 @@ public FraudCheckResult performFraudCheck(PersonIdentity personIdentity, boolean
             HttpResponse<String> httpResponse = sendHTTPRequestRetryIfAllowed(request);
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java:129]: Im most interested in the exact time experian takes. Can we start the instant just above this line and count it just after

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java [hunk 2]
```diff
@@ -143,6 +147,8 @@ public FraudCheckResult performFraudCheck(PersonIdentity personIdentity, boolean
...
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java:147]: Im most interested in the exact time experian takes. Can we start the instant just above this line and count it just after

File: lib/src/main/java/uk/gov/di/ipv/cri/fraud/library/metrics/Definitions.java [hunk 3]
```diff
@@ -65,6 +65,12 @@ public class Definitions {
     public static final String THIRD_PARTY_PEP_RESPONSE_TYPE_UNKNOWN =
             "third_party_pep_response_type_unknown";
 
+    // Third Party Response Latency
+    public static final String THIRD_PARTY_FRAUD_RESPONSE_LATENCY =
```
Comment (gds-jbentham) [inline lib/src/main/java/uk/gov/di/ipv/cri/fraud/library/metrics/Definitions.java:69]: can we include the time metric in the name like third_party_fraud_response_latency_millis

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java (hunk 1 above)
Comment (MMcC-GDS) [reply lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java:129]: Corrected.

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java (hunk 2 above)
Comment (MMcC-GDS) [reply lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java:148]: Corrected.

File: lib/src/main/java/uk/gov/di/ipv/cri/fraud/library/metrics/Definitions.java (hunk 3 above)
Comment (MMcC-GDS) [reply lib/src/main/java/uk/gov/di/ipv/cri/fraud/library/metrics/Definitions.java]: Name changed, value left as is.

File: lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java [hunk 4]
```diff
@@ -139,10 +143,13 @@ public FraudCheckResult performFraudCheck(PersonIdentity personIdentity, boolean
             HttpRequest request = requestBuilder(requestBody, requestBodyHmac);
             eventProbe.counterMetric(THIRD_PARTY_REQUEST_CREATED);
 
+            var startCheck = clock.instant();
             LOGGER.info("Submitting fraud check request to third party...");
             HttpResponse<String> httpResponse = sendHTTPRequestRetryIfAllowed(request);
 
             FraudCheckResult fraudCheckResult = responseHandler(httpResponse, false);
+            double durationCheck = Duration.between(startCheck, clock.instant()).toMillis();
```
Comment (gds-jbentham) [inline lambdas/fraudcheck/src/main/java/uk/gov/di/ipv/cri/fraud/api/gateway/ThirdPartyFraudGateway.java:151]: Mappers still do a bunch can we move this up one line

**Review by gds-jbentham — APPROVED** [review]

**Pull request merged**